core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -1159,7 +1159,7 @@ public int getSize() {
   }
 
   private static boolean isEqual(byte[] a1, byte[] a2) {
-    if (a1 == a2)
+    if (Arrays.equals(a1, a2))
```

Comment:
I'm not sure this keeps the same behaviour. The == will test if the passed arrays are the same reference.  Replacing with Arrays.equals is also going to check if the contents are the same.  This would make the rest of the test redundant - and from the code comments is trying to optimize the comparison.

Reply:
Thanks for the catch Ed! I will revert that change. It didn't really go with the rest of the changes anyway.